src/ui/RepoView.cpp

```diff
@@ -1056,6 +1056,15 @@ QFuture<git::Result> RepoView::fetch(
     RemoteCallbacks::Receive, entry, url, remote.name(), mWatcher, mRepo);
   connect(mCallbacks, &RemoteCallbacks::referenceUpdated,
           this, &RepoView::notifyReferenceUpdated);
+  connect(mCallbacks, &RemoteCallbacks::credentialsCanceled, [this] {
+    git::Config config = mRepo.appConfig();
+    bool defaultValue = Settings::instance()->value("global/autofetch/enable").toBool();
+    if (config.value<bool>("autofetch.enable", defaultValue)) {
+      // Disable autofetch; don't prompt for credentials again.
+      config.setValue("autofetch.enable", false);
```

Comment:
Do we really want to persist this? Another option is to just stop the timer so that it won't ask for the password anymore in the current session, but when the application restarts it asks again (once)?

Reply:
Some people don't notice that the auto-fetch causes this dialog and just think it's a bug. If it asks again when restarted, it would lead to same misunderstanding.
Also, you can easily reactivate auto-fetch anytime in your repository settings, so I think this change is harmless.

Reply:
Ok fair enough. So it's not just about it being a constant bother (for users who know/understand), but also confusing for other users.